src/coreclr/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComActivator.cs

```diff
@@ -124,7 +124,7 @@ public static object GetClassFactoryForType(ComActivationContext cxt)
 
             if (!Path.IsPathRooted(cxt.AssemblyPath))
             {
-                throw new ArgumentException();
+                throw new ArgumentException(null, nameof(cxt));
```

Comment:
Passing null as the first argument is going to change the exception message:
- `throw new ArgumentException();` => `Value does not fall within the expected range.`
- `throw new ArgumentException(null, "foo");` => `System.ArgumentException: Exception of type 'System.ArgumentException' was thrown. Parameter name: foo`

Seems we should do better than that?  Within corelib maybe we pass in `SR.Arg_ArgumentException`? Or we could change the behavior of ArgumentException's constructor to do:
```C#
            : base(message ?? SR.Arg_ArgumentException)
```
which in a way would be consistent with the constructors when no message is provided?

Reply:
We could also improve `ArgumentException.SetMessageField()` such that if the message is `SR.Arg_ArgumentException` and `ParamName` is set then it instead of `Value does not fall within the expected range.` uses `Value of '{0}' does not fall within the expected range.` 

Incidentally `Value does not fall within the expected range.` is not a great message because it's essentially the same message as ArgumentOutOfRangeException: `Specified argument was out of the range of valid values.`. It should probably be something like `Value is invalid.` or `Value of '{0}' is invalid."

Reply:
> throw new ArgumentException(null, "foo"); => System.ArgumentException: Exception of type 'System.ArgumentException' was thrown. Parameter name: foo

I see that is not happening from 3.0, i am getting message: `Value does not fall within the expected range. (Parameter 'foo')`. But see the other message in .Net Core 2.X and .Net Framework. So no issue with 3.0 the question is do we want to port the behavior to 2.X or .net framwork?

Reply:
> I see that is not happening from 3.0

You're right; I'm not sure what version I was running on, but I just tried on master and it's producing a value like you shared.  All good then. Thanks! (No need to port anything.)